backend/core/src/main/java/org/sonarsource/sonarlint/core/analysis/AnalysisService.java

```diff
@@ -223,12 +227,20 @@ private static org.sonarsource.sonarlint.core.rpc.protocol.common.Language toDto
   public GetAnalysisConfigResponse getAnalysisConfig(String configScopeId) {
     var bindingOpt = configurationRepository.getEffectiveBinding(configScopeId);
     var activeNodeJs = nodeJsService.getActiveNodeJs();
+    var inferredClientProperties = client.getInferredAnalysisProperties(new GetInferredAnalysisPropertiesParams(configScopeId)).join().getProperties();
+    clientAnalysisPropertiesRepository.setInferredProperties(configScopeId, inferredClientProperties);
+    var clientAnalysisProperties = clientAnalysisPropertiesRepository.getProperties(configScopeId);
 
     var nodeJsDetailsDto = activeNodeJs == null ? null : new NodeJsDetailsDto(activeNodeJs.getPath(), activeNodeJs.getVersion().toString());
-    return bindingOpt.map(binding -> new GetAnalysisConfigResponse(buildConnectedActiveRules(binding),
-      storageService.binding(binding).analyzerConfiguration().read().getSettings().getAll(), nodeJsDetailsDto,
-      Set.copyOf(pluginsService.getConnectedPluginPaths(binding.getConnectionId()))))
-      .orElseGet(() -> new GetAnalysisConfigResponse(buildStandaloneActiveRules(), Map.of(), nodeJsDetailsDto, Set.copyOf(pluginsService.getEmbeddedPluginPaths())));
+    return bindingOpt.map(binding -> {
+        var serverProperties = storageService.binding(binding).analyzerConfiguration().read().getSettings().getAll();
+        var analysisProperties = new ConcurrentHashMap<>(serverProperties);
```

Comment:
Do you really need a concurrent hash map here?

Reply:
Maybe not. Changed to HashMap.